### PR TITLE
Rejigger home page markup and styles

### PR DESCRIPTION
**Pull request by tysongach**
I apologize in advance for such a large PR - I’m normally one to try and 
make PRs as small as possible 😅 

While starting to tackle the [creation of linter/language landing pages](https://trello.com/c/QgdIrLL6/907-create-individual-language-linter-pages-linked-from-landing-page)
I found the CSS to be working against me more than being helpful. I started 
making small improvements to help me but quickly realized that for such a 
small website it’d be quickest to tackle this in one big set of changes.

The design is mostly the same with a few exceptions and improvements, 
which I’ve highlighted below.

---

What’s done in this PR…

- Organize styles using [ITCSS](https://www.creativebloq.com/web-design/manage-large-css-projects-itcss-101517528) (inverted triangle CSS) and use
  class namespaces (i.e. `c-` for components, `o-` for objects)
- Detach styles from being named after the ‘home’ page - making them
  more reusable and descriptive of their purpose
- Leverage CSS Custom Properties for better color management and
  component variants
    - This also opens the door for spinning-up themed landing pages,
      where the design can be unique to linter, for example
- New styles do not need the Bourbon and Neat Sass libraries, so we
  could potentially drop these dependencies
- Add and use the active_link_to gem which improves the accessibility of
  navigation links by outputting the `aria-current` attribute on the
  current page link
- Use Rails Internationalization
- Use WebP images (progressively-enhanced)
- Improve accessibility with better heading hierarchy, added image alt
  text, and labeling page landmarks
- Improve the Pricing section design

---

## Highlighted changes 

See the whole thing here: https://staging.houndci.com

### Hero before

Screen Shot 2020-08-15 at 11 40 48

### Hero after

Tightened-up and slightly heavier typography

Screen Shot 2020-08-15 at 11 40 53

### Pricing section before

I found this section hard to parse for how little info it needed to get across:

- The open source section and card is fully gray, but the private repo cards were a mix of purple and gray - being more consistent makes the whole thing easier on the eyes to comprehend that color means something here
- Duplicative ‘Open Source’ and ‘Public Repos’ text
- Colors felt washed out

pricing-old

### Pricing section after

- Cards are in order starting with the cheapest
- The whole section is much smaller

pricing-new

### Configuration section before

The alignment felt off here with the rest of the page and I didn’t think the graphic added much, so simplified this to be just text.

config-old

### Configuration section after

config-new

### Code quality section before

We use flat colors everywhere, so this gradient felt out of place.

Screen Shot 2020-08-15 at 11 33 43-fullpage copy

### Code quality section after

Screen Shot 2020-08-15 at 11 37 58-fullpage copy

### CODE REVIEW

File: app/assets/stylesheets/marketing.scss [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+@import "marketing/components/button";
+@import "marketing/components/page-hero";
+@import "marketing/components/page-section";
+@import "marketing/components/plan";
+@import "marketing/components/text-partition";
+
+@import "marketing/utilities/font-weight";
+@import "marketing/utilities/line-height";
+@import "marketing/utilities/margin";
+@import "marketing/utilities/text-align";
```
Comment (tysongach) [inline app/assets/stylesheets/marketing.scss:29]: This is a new and separate Sass manifest from the application manifest - it’s meant to be a temporary solution and these can be folded back into one manifest when we can modernize the application styles.

File: app/assets/stylesheets/pages/_pages.scss [hunk 2]
```diff
@@ -3,4 +3,3 @@
 @import "builds";
 @import "docs";
 @import "tier_change";
-@import "home/home";
```
Comment (tysongach) [inline app/assets/stylesheets/pages/_pages.scss:6]: No more page-based styles for the home page - all the new styles are meant to be reusable, composable, and more widely applicable.

File: app/controllers/home_controller.rb [hunk 3]
```diff
@@ -3,6 +3,8 @@ class HomeController < ApplicationController
 
   before_action :redirect_to_repos, if: :signed_in?
 
+  layout "marketing"
```
Comment (tysongach) [inline app/controllers/home_controller.rb:6]: This is a temporary solution - we can get back to one layout, too.

File: app/views/home/_pricing.html.erb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        <li class="o-grid__item o-grid__item--span-3@medium c-plan c-plan--private">
+          <p class="c-plan__title">Labrador</p>
+          <p class="c-plan__allowance">Up to <strong>10 private</strong> repos</p>
+          <p class="c-plan__price"><span class="c-plan__dollar-amount">$99</span> month</p>
+        </li>
+        <li class="o-grid__item o-grid__item--span-3@medium c-plan c-plan--private">
+          <p class="c-plan__title">Great Dane</p>
+          <p class="c-plan__allowance">Up to <strong>30 private</strong> repos</p>
+          <p class="c-plan__price"><span class="c-plan__dollar-amount">$249</span> month</p>
+        </li>
```
Comment (tysongach) [inline app/views/home/_pricing.html.erb:28]: I couldn’t figure out how to use the existing `plans` partials and use the new CSS classes - so this is some duplication, which is unfortunate. I think it’s fine but we should definitely look to leverage those shared partials again.

File: app/views/home/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+<%= render partial: "hero" %>
+<%= render partial: "companies", locals: { companies: @companies } %>
+<%= render partial: "languages", locals: { languages: @languages } %>
+<%= render partial: "code_quality" %>
+<%= render partial: "how_it_works" %>
+<%= render partial: "review" %>
+<%= render partial: "guides" %>
+<%= render partial: "pricing" %>
+<%= render partial: "security" %>
```
Comment (tysongach) [inline app/views/home/index.html.erb:12]: With each section of the page in their own partial, we can more easily reuse, remix, and try different things.

File: app/views/layouts/marketing.html.erb [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      rel: "preload",
+      type: "font/woff",
+    ) %>
+    <%= tag.link(
+      as: "font",
+      crossorigin: "anonymous",
+      href: "https://font.thoughtbot.com/hound/Tofino-Medium.woff",
+      rel: "preload",
+      type: "font/woff",
+    ) %>
```
Comment (tysongach) [inline app/views/layouts/marketing.html.erb:19]: Preload fonts for a performance boost.

File: app/views/home/_how_it_works.html.erb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+          type: "image/png",
+        ) %>
+
+        <%= image_tag(
+          "home/hound_flow.png",
+          alt: t(".image_alt"),
+          loading: "lazy",
+          size: "1798x496",
+        ) %>
+      </picture>
```
Comment (tysongach) [inline app/views/home/_how_it_works.html.erb:26]: Using the `<picture>` element to progressively-enhance to a WebP image - it’s a smaller file size so we get a small performance boost. Also leveraging native lazy loading of the image.

File: app/views/application/_footer.haml [hunk 8]
```diff
@@ -17,7 +17,7 @@
       new_window_options(class: "app-footer__nav-link")
 
   %p
-    Questions? Try
-    #{link_to "Help", ENV.fetch("HELP_URL"), new_window_options}
+    Questions? Try our
+    #{link_to "Help Center", ENV.fetch("HELP_URL"), new_window_options}
```
Comment (salbertson) [inline app/views/application/_footer.haml:21]: Can we use the new translation here?

**Review by salbertson — APPROVED** [review]
@tysongach this looks great! There's a lot here, I only have one minor question. I deployed to Staging and looked good to me. 👍

File: app/views/application/_footer.haml (hunk 8 above)
Comment (tysongach) [reply app/views/application/_footer.haml:21]: Yes! Though, I’d like to tackle this in a future PR. Here I was doing a bit of separation with marketing, so wanted to focus on that. I hope to work our way back to shared partials and with that, everything can use translations.
